src/python/pants/engine/native.py

```diff
@@ -327,6 +327,15 @@ def extern_type_to_str(self, context_handle, type_id):
     c = self._ffi.from_handle(context_handle)
     return c.utf8_buf(text_type(c.from_id(type_id.tup_0).__name__))
 
+  @_extern_decl('Buffer', ['ExternContext*', 'Handle*'])
+  def extern_val_to_bytes(self, context_handle, val):
+    """Given a Handle for `obj`, write bytes(obj) and return it."""
+    c = self._ffi.from_handle(context_handle)
+    v = c.from_value(val[0])
+    # Consistently use the empty string to indicate None.
+    v_bytes = b'' if v is None else binary_type(v)
```

Comment:
Hm. Was this case propagated from the string handling? Can we avoid adding it here and see what happens? Would be nice to allow for it if it isn't needed.

Reply:
Since we do it with the Unicode handling right below, I think it's appropriate to do here too. It would be surprising to only do with one stringy thing and not the other.

As a followup, we could rethink these two lines.

Reply:
Automatically transforming nulls like this is an anti-feature, imo. It might have been helpful in the slightly more general case of strings, but null byte fields would just be strange.

Reply:
Yeah, this was copied and pasted. Removed.